libs/contrib/Interfaces/Verified.idr

```diff
@@ -40,49 +69,118 @@ interface (Monad m, VerifiedApplicative m) => VerifiedMonad (m : Type -> Type) w
   monadAssociativity : (mx : m a) -> (f : a -> m b) -> (g : b -> m c) ->
                        (mx >>= f) >>= g = mx >>= (\x => f x >>= g)
 
+VerifiedMonad Maybe where
+    monadApplicative Nothing Nothing = Refl
+    monadApplicative Nothing (Just x) = Refl
+    monadApplicative (Just x) Nothing = Refl
+    monadApplicative (Just x) (Just y) = Refl
+    monadLeftIdentity x f = Refl
+    monadRightIdentity Nothing = Refl
+    monadRightIdentity (Just x) = Refl
+    monadAssociativity Nothing f g = Refl
+    monadAssociativity (Just x) f g = Refl
+
+-- TODO VerifiedMonad instances for: List, Stream, Either e
 
 interface Semigroup a => VerifiedSemigroup a where
   total semigroupOpIsAssociative : (l, c, r : a) -> l <+> (c <+> r) = (l <+> c) <+> r
 
 implementation VerifiedSemigroup (List a) where
   semigroupOpIsAssociative = appendAssociative
 
---implementation VerifiedSemigroup Nat where
---  semigroupOpIsAssociative = plusAssociative
+[PlusNatSemiV] VerifiedSemigroup Nat using PlusNatSemi where
+  semigroupOpIsAssociative = plusAssociative
 
+[MultNatSemiV] VerifiedSemigroup Nat using MultNatSemi where
+  semigroupOpIsAssociative = multAssociative
+
+[PlusZZSemiV] VerifiedSemigroup ZZ using PlusZZSemi where
+  semigroupOpIsAssociative = plusAssociativeZ
+
+[MultZZSemiV] VerifiedSemigroup ZZ using MultZZSemi where
+  semigroupOpIsAssociative = multAssociativeZ
+
+-- TODO Verified versions of...
+--  (Semigroup m, Semigroup n) => Semigroup (m, n) where
+--    (a, b) <+> (c, d) = (a <+> c, b <+> d)
+--
+-- Semigroup (Maybe a)
+--
+-- [collectJust] Semigroup a => Semigroup (Maybe a)
+--
+-- libs/base/Data/Morphisms.idr: implementation Monoid (Endomorphism a)
 
 interface (VerifiedSemigroup a, Monoid a) => VerifiedMonoid a where
   total monoidNeutralIsNeutralL : (l : a) -> l <+> Algebra.neutral = l
   total monoidNeutralIsNeutralR : (r : a) -> Algebra.neutral <+> r = r
 
--- implementation VerifiedMonoid Nat where
---   monoidNeutralIsNeutralL = plusZeroRightNeutral
---   monoidNeutralIsNeutralR = plusZeroLeftNeutral
+[PlusNatMonoidV] VerifiedMonoid Nat using PlusNatSemiV, PlusNatMonoid where
+   monoidNeutralIsNeutralL = plusZeroRightNeutral
+   monoidNeutralIsNeutralR = plusZeroLeftNeutral
+
+[MultNatMonoidV] VerifiedMonoid Nat using MultNatSemiV, MultNatMonoid where
+  monoidNeutralIsNeutralL = multOneRightNeutral
+  monoidNeutralIsNeutralR = multOneLeftNeutral
+
+[PlusZZMonoidV] VerifiedMonoid ZZ using PlusZZSemiV, PlusZZMonoid where
+   monoidNeutralIsNeutralL = plusZeroRightNeutralZ
+   monoidNeutralIsNeutralR = plusZeroLeftNeutralZ
+
+[MultZZMonoidV] VerifiedMonoid ZZ using MultZZSemiV, MultZZMonoid where
+  monoidNeutralIsNeutralL = multOneRightNeutralZ
+  monoidNeutralIsNeutralR = multOneLeftNeutralZ
 
 implementation VerifiedMonoid (List a) where
   monoidNeutralIsNeutralL = appendNilRightNeutral
   monoidNeutralIsNeutralR xs = Refl
 
+-- TODO Verified versions of...
```

Comment:
Could we kindly remove the 'TODO' comments from the library code, or resolve them? I think it is not nice to have code that is commented out in the libraries.